kentikapi/users_integration_test.go

```diff
@@ -598,6 +602,366 @@ func TestClient_DeleteUser(t *testing.T) {
+			user:                models.User{},
+			expectedRequestBody: newEmptyUserRequestBody(),
+			retryMax: intPtr(5),
+			responses:      []httpResponse{
+				newErrorHTTPResponse(http.StatusInternalServerError),
+				newErrorHTTPResponse(http.StatusBadGateway),
+				newErrorHTTPResponse(http.StatusServiceUnavailable),
+				newErrorHTTPResponse(http.StatusGatewayTimeout),
+				newErrorHTTPResponse(http.StatusTooManyRequests),
+				newErrorHTTPResponse(http.StatusBadGateway),
+			},
+			expectedError:       true,
+		},
+		{
+			name: "retry till success when status 429 too many requests received",
+			user: *models.NewUser(models.UserRequiredFields{
+				Username:     "testuser",
+				UserFullName: "Test User",
+				UserEmail:    "test@user.example",
+				Role:         "Member",
+				EmailService: true,
+				EmailProduct: false,
+			}),
+			expectedRequestBody: object{
+				"user": object{
+					"username":       "testuser",
+					"user_full_name": "Test User",
+					"user_email":     "test@user.example",
+					"role":           "Member",
+					"email_service":  true,
+					"email_product":  false,
+				},
+			},
+			responses: []httpResponse{
+				newErrorHTTPResponse(http.StatusTooManyRequests),
+				newErrorHTTPResponse(http.StatusTooManyRequests),
+				{http.StatusCreated,
+					`{
+						"user": {
+							"id": "145999",
+							"username": "testuser",
+							"user_full_name": "Test User",
+							"user_email": "test@user.example",
+							"role": "Member",
+							"email_service": "true",
+							"email_product": "false",
+							"last_login": null,
+							"created_date": "2020-12-09T14:48:42.187Z",
+							"updated_date": "2020-12-09T14:48:43.243Z",
+							"company_id": "74333",
+							"user_api_token": null,
+							"filters": {},
+							"saved_filters": []
+						}
+					}`,
+				},
+			},
+			expectedResult: &models.User{
+				ID:           145999,
+				Username:     "testuser",
+				UserFullName: "Test User",
+				UserEmail:    "test@user.example",
+				Role:         "Member",
+				EmailService: true,
+				EmailProduct: false,
+				LastLogin:    nil,
+				CreatedDate:  *testutil.ParseISO8601Timestamp(t, "2020-12-09T14:48:42.187Z"),
+				UpdatedDate:  *testutil.ParseISO8601Timestamp(t, "2020-12-09T14:48:43.243Z"),
+				CompanyID:    74333,
+				UserAPIToken: nil,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// arrange
+			h := newSpyHTTPHandler(t, tt.responses)
+			s := httptest.NewServer(h)
+			defer s.Close()
+
+			c := kentikapi.NewClient(kentikapi.Config{
+				APIURL:    s.URL,
+				AuthEmail: dummyAuthEmail,
+				AuthToken: dummyAuthToken,
+				RetryCfg: utils.RetryConfig{
+					MaxAttempts:          tt.retryMax,
+					MinDelay:             durationPtr(1 * time.Microsecond),
+					MaxDelay:             durationPtr(10 * time.Microsecond),
+					RetryableStatusCodes: tt.retryStatusCodes,
+					RetryableMethods:     tt.retryMethods,
+				},
+			})
+
+			// act
+			result, err := c.Users.
```

Comment:
IMO redundant line break

Reply:
I will look into it while configuring linters.